rfcs/inline_symbols.md

```diff
@@ -0,0 +1,624 @@
+ |  |  +---------------------------- 0b1010_0101 / VarInt 37 / symbol ID field name ($37)
+ |  +------------------------------- VarUInt Length: 9 bytes
+ +---------------------------------- Type descriptor: inline symbol struct
+```
+
+If the sign bit is positive and the magnitude is zero, the field name is symbol ID `0`.
+
+If the sign bit is negative and the magnitude is zero, the field name is the empty string.
+
+The [existing rules for NOP padding in struct
+fields](http://amzn.github.io/ion-docs/docs/binary.html#nop-padding-in-struct-fields) also apply to
+this representation.
+
+#### Inline annotations
+
+Ion 1.0's [existing encoding for
+annotations](http://amzn.github.io/ion-docs/docs/binary.html#annotations) uses a 'wrapper' to
+associate a set of annotations with a given value. The wrapper contains a sequence of annotation
+symbol IDs followed by the value being annotated. The wrapper's header provides separate length
+fields for both:
+
+1. **The total size of the encoded symbol ID sequence and the encoded value.** This allows the
+   reader to skip over the entire wrapper if needed, moving to the next value in the Ion stream.
+2. **The size of the encoded symbol ID sequence.** This allows the reader to skip over the
+   annotations, moving to the annotated value inside the wrapper.
+
+For example, this value:
+```js
+Author::"Ernest Hemingway"
+```
+would be encoded in binary Ion as:
+```
+                   E  r  n  e  s  t     H  e  m  i  n  g  w  a  y
+EE 94 81 8A 8E 90 45 72 6E 65 73 74 20 48 65 6D 69 6E 67 77 61 79 
+ ^  ^  ^  ^  ^  +---- Length: 16 bytes
+ |  |  |  |  +------- String w/VarUInt Length
+ |  |  |  +---------- Annotation symbol ID $10
+ |  |  +------------- Annotations Length: 1 byte
+ |  +---------------- Annotations+Value Length: 20 bytes 
+ +------------------- Annotation wrapper w/VarUInt Length
+```
+*(The above assumes that 'Author' is already in the active symbol table as `$10`.)*
+
+
+In practice, few readers leverage this capability due in part to their need to read the wrapped
+value's type descriptor byte before deciding whether the value can be skipped. Additionally, the
+wrapped values themselves already specify their encoded size. This means that two of the four
+Annotation header bytes in the above example are of dubious value.
+
+This RFC adds two new encodings for annotations:
+1. An encoding that is optimized for the most common case, in which there is a single annotation.
+2. An encoding that is optimized for the case in which a value has multiple annotations.
+
+Both encodings support inline symbol definitions using the `VarInt` encoding scheme described in
+the section [*Inline symbol structs*](#0xf4-inline-symbol-structs).
+
+Because Ion 1.0's wrapper encoding cannot be shorter than 3 bytes, type descriptor bytes `0xE1` and
+`0xE2` were not legal type descriptor bytes. We use them in Ion 1.1 to represent our new encodings.
+
+##### `0xE1`: Singleton inline symbol annotation
```

Comment:
Maybe better phrased as "Single, optionally inline annotation" due to the modality of the encoding.

Reply:
Per offline discussion, I'm using the phrase `inlineable`.